File: connectors/connector-redmine/src/main/scala/com/taskadapter/connector/redmine/CustomFieldBuilder.scala
package com.taskadapter.connector.redmine

import java.util

import com.taskadapter.model.Field
import com.taskadapter.redmineapi.bean.{CustomFieldDefinition, CustomFieldFactory, Issue}

// TODO move to Redmine Java API?
object CustomFieldBuilder {

  def add(issue: Issue, customFieldDefinitions: util.List[CustomFieldDefinition], field: Field[_], value: String): Unit = {
    val customFieldId = CustomFieldDefinitionFinder.findCustomFieldId(customFieldDefinitions, field)
    val customField = CustomFieldFactory.create(customFieldId, field.getFieldName(), value.toString)
    issue.addCustomField(customField)
  }
}

File: connectors/connector-redmine/src/main/java/com/taskadapter/connector/redmine/CustomFieldBuilder.java
package com.taskadapter.connector.redmine;

import com.taskadapter.model.Field;
import com.taskadapter.redmineapi.bean.CustomFieldDefinition;
import com.taskadapter.redmineapi.bean.CustomFieldFactory;
import com.taskadapter.redmineapi.bean.Issue;

import java.util.List;

// TODO move to Redmine Java API?
public class CustomFieldBuilder {

    public static void add(Issue issue, List<CustomFieldDefinition> customFieldDefinitions, Field<?> field, String value) {
        var customFieldId = CustomFieldDefinitionFinder.findCustomFieldId(customFieldDefinitions, field);
        var customField = CustomFieldFactory.create(customFieldId, field.getFieldName(), value);
        issue.addCustomField(customField);
    }
}
